python/mxnet/kvstore.py

```diff
@@ -379,12 +400,33 @@ def _send_command_to_servers(self, head, body):
 def create(name='local'):
     """Creates a new KVStore.
 
+    For single machine training, there are two commonly used types:
+
+    ``local``: all gradients are copied to CPU memory and weights are updated there.
+
+    ``device``: both gradient aggregation and weight updates are run on GPUs. With this setting,
+    the KVStore also attempts to use GPU peer-to-peer communication,
+    potentially accelerating the communication.
+
+    For distributed training, KVStore also supports a number of types:
+
+    ``dist_sync``: it behaves similarly to ``local`` but exhibits one major difference.
+    With ``dist_sync``, batch-size now means the batch size used on each machine.
+    So if there are n machines and we use batch size b,
+    then ``dist_sync`` behaves like local with batch size n * b.
+
+    ``dist_device_sync``: it is identical to ``dist_sync`` with the difference similar
+    to ``device`` vs ``local``.
+
+    ``dist_async`` performs asynchronous updates.
+    The weight is updated whenever gradients are received from any machine. The update is atomic,
```

Comment:
weight is -> weights are

"update is atomic": You mean gradient updates are serialized?